packages/lodestar/src/eth1/provider/jsonRpcHttpClient.ts

```diff
@@ -75,6 +77,23 @@ export class JsonRpcHttpClient implements IJsonRpcHttpClient {
     return parseRpcResponse(res, payload);
   }
 
+  /**
+   * Perform RPC request with retry
+   */
+  async fetchWithRetries<R, P = IJson[]>(payload: IRpcPayload<P>, opts?: ReqOpts): Promise<R> {
+    const res: IRpcResponse<R> = await retry(
+      async (_attempt) => {
+        return await this.fetchJson({jsonrpc: "2.0", id: this.id++, ...payload}, opts);
+      },
+      {
+        retries: opts?.retryAttempts ?? 1,
+        retryDelay: opts?.retryDelay ?? 0,
+        shouldRetry: opts?.shouldRetry,
+      }
+    );
+    return parseRpcResponse(res, payload);
```

Comment:
The should retry logic here should be based exclusively in networking errors. You only want to retry when the EL is unavailable. Here it's a bit tough because the underlying httpclient can change. So you should add a lot of unit (or e2e) tests that spin up a real server and try different things like:
- bad URL
- bad port
- NGINX rejects to forward
- etc

And ensure that only those are retried. Otherwise you can try to guess when a response is actually from an EL client, and don't retry only when you know the error is an "app layer" EL error

Reply:
> The should retry logic here should be based exclusively in networking errors. You only want to retry when the EL is unavailable

I believe this the case already. Line 94 (`parseRpcResponse`) would only be reached if the call returns 200, and  `parseRpcResponse` parses that and throws `ErrorJsonRpcResponse`. If the call fails for any other reason (if response is not ok, ie network errors) that is only when the request is retried. 

> So you should add a lot of unit (or e2e) tests that spin up a real server 

I'll add these